config/application.rb

```diff
@@ -40,6 +40,15 @@ class Application < Rails::Application
       Clients::Vault.token = Config[:vault_token]
       Clients::Vault.configure_kv
       Clients::Vault.configure_pki
+      issuer = config.astral.oidc_provider[:issuer]
+      client_id = config.astral.oidc_provider[:client_id]
+      client_secret = config.astral.oidc_provider[:client_secret]
+      if config.astral.configure_oidc_provider?
```

Comment:
It feels really weird to me to have app logic in a config file. IMO this change and everything surrounding it here belongs in an initializer, but at minimum can we refactor to remove the "if"?

Reply:
If you're trying to do it per-environment, Rails should support that in config/{environment}.rb, that might be simplest for now.

Reply:
> everything surrounding it here

when you say "everything", do you mean the lines that Geoff added before my changes as well?: https://github.com/G-Research/astral/blob/5561da973fd0d628e3e1a18ab6db22216615cfc1/config/application.rb#L40-L43

Reply:
done